components/resolved_ts/src/advance.rs

```diff
@@ -409,21 +405,8 @@ impl LeadershipResolver {
                     }
                 }
             }
-        }
-        for (region_id, prs) in region_map {
-            if prs.is_empty() {
-                // The peer had the leadership before, but now it's no longer
-                // the case. Skip checking the region.
-                continue;
-            }
-            if let Some(resp) = resp_map.get(region_id) {
-                if resp.is_empty() {
-                    // No response, maybe the peer lost leadership.
-                    continue;
-                }
-                if region_has_quorum(prs, resp) {
-                    valid_regions.insert(*region_id);
-                }
+            if valid_regions.len() >= progresses.len() {
```

Comment:
And I think we can add a log at the end of this function when `valid_regions.len()` is not equal to `checking_regions.len()`. It's useful to debug some resolve ts not moving forward issue.

Reply:
Done!